mathics_scanner/prescanner.py

```diff
@@ -58,6 +58,8 @@ def scan(self):
                         self.incomplete()
                     self.stubs.append(self.code[self.start : self.pos])
                     self.newstub(self.pos + 2)
+                elif c == "\\":
+                    self.pos += 2
```

Comment:
To say the truth I don't know why it works - maybe it doesn't sometimes -, basically here if a backslash is found the next character is ignored, and that seems very problematic.

Reply:
No, the idea is that if a backslash is followed by another backslash, this second backslash should not be considered an escape character by the prescanner, so the next character to be analyzed is the one after the second backslash.

Reply:
@TiagoCavalcante. I have added a comment trying to explain why it works, and I have simplified the logic.

Reply:
> To say the truth I don't know why it works - maybe it doesn't sometimes -, basically here if a backslash is found the next character is ignored, and that seems very problematic.

Here is how that "while" loop works, `self.pos` is a cursor that scans over `self.code` accumulating list `self.stubs`.
 
When it finds something "interesting" it stores in `self.stubs` the previous text and a replacement text for an escape sequence that it found. For example, for  `\[Integral]` I guess that text is replaced with a single character. But I see there are also sequences that start `\:xxx` and `\octal-number`. 

When we see two backslashes that is uninteresting so the cursor just advances over these and this text is accumulated in the current self.subs. What is weird though is that somehow Python's backslash escaping is turning the two backslashes into one.